StatePrinter.Tests/IntegrationTests/DictionaryTest.cs

```diff
@@ -168,219 +211,263 @@ public void Dictionary_int_int()
     }
-    Value = ""6""
-    ReferenceNo = null
-}
-[13] = new Token(), ref: 5
-{
-    Tokenkind = EndScope
-    FieldType = null
-    Field = null
-    Value = null
-    ReferenceNo = null
-}
-[14] = new Token()
-{
-    Tokenkind = FieldnameWithTypeAndReference
-    FieldType =  -> 2
-    Field = new Field()
+    new Token()
     {
-        Name = ""untyped_int_int""
-        SimpleKeyInArrayOrDictionary = ""1""
+        Tokenkind = FieldnameWithTypeAndReference
+        FieldType = -> 1
+        Field = new Field()
+        {
+            Name = ""untyped_int_int""
+            Key = null
+            Index = 1
+        }
+        Value = null
+        ReferenceNo = new Reference()
+        {
+            Number = 4
+        }
     }
-    Value = null
-    ReferenceNo = new Reference()
+    -> 0
+    new Token()
     {
-        Number = 4
+        Tokenkind = SimpleFieldValue
+        FieldType = null
+        Field = -> 2
+        Value = ""2""
+        ReferenceNo = null
     }
-}
-[15] =  -> 1
-[16] = new Token()
-{
-    Tokenkind = SimpleFieldValue
-    FieldType = null
-    Field =  -> 3
-    Value = ""2""
-    ReferenceNo = null
-}
-[17] = new Token()
-{
-    Tokenkind = SimpleFieldValue
-    FieldType = null
-    Field =  -> 4
-    Value = ""4""
-    ReferenceNo = null
-}
-[18] =  -> 5
-[19] = new Token()
-{
-    Tokenkind = FieldnameWithTypeAndReference
-    FieldType =  -> 2
-    Field = new Field()
+    new Token()
     {
-        Name = ""untyped_int_int""
-        SimpleKeyInArrayOrDictionary = ""2""
+        Tokenkind = SimpleFieldValue
+        FieldType = null
+        Field = -> 3
+        Value = ""4""
+        ReferenceNo = null
     }
-    Value = null
-    ReferenceNo = new Reference()
+    -> 4
+    new Token()
     {
-        Number = 5
+        Tokenkind = FieldnameWithTypeAndReference
+        FieldType = -> 1
+        Field = new Field()
+        {
+            Name = ""untyped_int_int""
+            Key = null
+            Index = 2
+        }
+        Value = null
+        ReferenceNo = new Reference()
+        {
+            Number = 5
+        }
     }
-}
-[20] =  -> 1
-[21] = new Token()
-{
-    Tokenkind = SimpleFieldValue
-    FieldType = null
-    Field =  -> 3
-    Value = ""1""
-    ReferenceNo = null
-}
-[22] = new Token()
-{
-    Tokenkind = SimpleFieldValue
-    FieldType = null
-    Field =  -> 4
-    Value = ""2""
-    ReferenceNo = null
-}
-[23] =  -> 5
-[24] =  -> 6
-[25] =  -> 5";
+    -> 0
+    new Token()
+    {
+        Tokenkind = SimpleFieldValue
+        FieldType = null
+        Field = -> 2
+        Value = ""1""
+        ReferenceNo = null
+    }
+    new Token()
+    {
+        Tokenkind = SimpleFieldValue
+        FieldType = null
+        Field = -> 3
+        Value = ""2""
+        ReferenceNo = null
+    }
+    -> 4
+    new Token()
+    {
+        Tokenkind = EndList
+        FieldType = null
+        Field = null
+        Value = null
+        ReferenceNo = null
+    }
+    -> 4
```

Comment:
This is correct. The list contains different elements now and the reference numbering is different.